### ISSUE TITLE
Game crashes randomly when editing a train route

### ISSUE THREAD

**Issue by Lev135** [OpenLoco/OpenLoco#2267]
**Version information:**
 - Platform: Windows 10
 - Version: 24.01.1
 - Build/commit: 839d396

**Describe the bug**
Sometimes game crashes when I select a route item/add new one. This happens randomly, but often enough. No report is produced, so I don't know, what to attach to the issue. 

It seems like crashes are more often in complex routes (with 5+ stations) with trains, but sometimes with simple routes too. For other vehicles everything is fine.

**To Reproduce**
I don't know what exectly one should do to crash the game. 

**Expected behaviour**
I want the game not to crush ^).

**Screenshots**
![image](https://github.com/OpenLoco/OpenLoco/assets/35816057/5d842590-c172-4c3d-ab0c-5029d2d4abda)

**Additional context**

**Comment by ilmoro93** [comment]
I have this problem very often.

For example in this saved game (version 24.02.1): [North America (Midwest) - 100 Year Challenge - 1.zip](https://github.com/OpenLoco/OpenLoco/files/14594108/North.America.Midwest.-.100.Year.Challenge.-.1.zip)

Steps to reproduce the behaviour:

1. Remove train 14 from landscape
2. Delete all orders/stops from route
3. Add waypoint 1 as the screenshot below:

![Screenshot (3)](https://github.com/OpenLoco/OpenLoco/assets/56093089/1bf73acb-d61f-45fd-8f9c-27044cb3adbe)

4. Add stop to Dallas Transfer
5. Wait some seconds... 

This is just an example. As said before, it happens frequently with complex routes, but it has happened also with buses.

Another strange thing here: on the right of waypoint some kinds of cargo have appeared randomly (for example grain, that was not related to that train or track).

![image](https://github.com/OpenLoco/OpenLoco/assets/56093089/6c666ed9-0154-4021-a932-a893779dfe36)

**Comment by AaronVanGeffen** [comment]
@ilmoro93 Are you actually on v24.02.1, or on master? I ask because #2327 (in master) introduced a regression in dropdowns, which is fixed by #2339 (not yet in master).

**Comment by ilmoro93** [comment]
Sorry, in debug I was in my branch (https://github.com/ilmoro93/OpenLoco/tree/restart-music-on-resume) which is one commit on #2327.

**Comment by duncanspumpkin** [comment]
> I tried to reproduce my last comment in debug. After adding stop to Dallas Transfer I pressed on + symbol to add an order and I got this exception:
> 
> ```
> Exception thrown: read access violation.
> **OpenLoco::Gfx::getG1Element**(...) returned nullptr.
> ```
> 
> here in `Openloco::Drawing::Impl:`
> 
> ```
> case ControlCodes::inlineSpriteStr:
> {
> 	const uint32_t image = reinterpret_cast<const uint32_t*>(str)[0];
> 	const uint32_t imageId = image & 0x7FFFF;
> 	str += 4;
> 	width += Gfx::getG1Element(imageId)->width;
> 	break;
> }
> ```

Was that on the latest master version? Or on 24.02.1 debug. There is a bug on master that crashes when you add orders it's unrelated to this random crash though.

**Comment by ilmoro93** [comment]
It was on this commit in my branch https://github.com/ilmoro93/OpenLoco/commit/95e65373c83f9f39648caaacf142e378b03bf369, which is basically this commit on master https://github.com/OpenLoco/OpenLoco/commit/d52f0c78cf4a613222398f440c53b6ec68e4bfb0 with one single change.

**Comment by ilmoro93** [comment]
Another crash in v24.02.1 (8e40251e on master).
[crash-edit-truck-route.zip](https://github.com/OpenLoco/OpenLoco/files/14620021/crash-edit-truck-route.zip)

When I edit the route of a running vehicle (truck for example) it crashes after some seconds.

**Comment by ilmoro93** [comment]
I cannot reproduce any of the crashes on f1abcfc5d94199c0eeb62eaaf8dd2c8d21f5d1cd in debug. Was there some fix after 8e40251ee3280d73ec4f54c01afbf284cc1274b8 regarding this issue?

**Comment by AaronVanGeffen** [comment]
That makes sense. As [mentioned previously](https://github.com/OpenLoco/OpenLoco/issues/2267#issuecomment-1995998325), #2327 introduced a regression in dropdowns, which was fixed by #2339, which was merged into master as f1abcfc5d94199c0eeb62eaaf8dd2c8d21f5d1cd.

**Comment by ilmoro93** [comment]
When you say that #2327 introduced a regression, are you referring to the bug in [this comment](https://github.com/OpenLoco/OpenLoco/issues/2267#issuecomment-1995976324)?

I was referring to the other crashes (when editing routes in general). I had trouble in reproducing them.

Now I have managed to make the game stuck (in f1abcfc5d94199c0eeb62eaaf8dd2c8d21f5d1cd). While pausing the code in debug it seems to loop indefinitely in this function, called by external code:

```
    OrderRingView::Iterator& OrderRingView::Iterator::operator++()
    {
        auto* newOrders = reinterpret_cast<uint8_t*>(_currentOrder) + kOrderSizes[static_cast<uint8_t>(_currentOrder->getType())];
        _currentOrder = reinterpret_cast<Order*>(newOrders);
        if (_currentOrder->getType() == OrderType::End)
        {
            _currentOrder = _beginOrderTable;
            _hasLooped = true;
        }
        return *this;
    }```
